### DISCUSSION TITLE
Pagination requests are not working in my extension, what am I missing?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#451] [Ask a question] [answered]
Hello, 

I'm currently writing an extension which displays the markdown source of a page (useful for wiki pages etc). My idea was to simply use the `OnParsePageLayout` function, and fetch the page source as pagination request: 

``` php
// Handle page layout
public function onParsePageLayout($page, $name) {
    $pagination = $this->yellow->system->get("contentPagination");
    if ($_REQUEST[$pagination]==$this->yellow->system->get("pagesourceFile") && $this->yellow->page->get("source")) {
        $this->yellow->page->setHeader("Content-Type", "text/plain; charset=utf-8");
        $title = $this->yellow->page->get("title");
        $titleLenght = strlenu($title);
        $output = $title."\r\n";
        $output .= str_pad("=", $titleLenght, "=")."\r\n";
        $output .= $this->yellow->page->getContent(true);
        $this->yellow->page->setOutput($output);
    }
}
```

This works perfectly, however only for the home page. Calling the pagination from another location (e. g. `https://example.com/subdir/page:source.md`), results in a "file not found" error. What am I missing here? Probably one of these tiny switches again. 😅

Steffen

**Comment by markseuffert** [comment, marked as answer]
Try a different location argument. If you use `page`  it may be handled by another extension for pagination and that's why you get a "file not found" error. Just a guess.

**Comment by schulle4u** [comment]
And it is the right guess. At least I'm now able to call the pagination for subdirectories, but not for pages inside subdirectories, e.g. `https://example.com/subdir/subpage`. Probably I have to make a workaround for page URLs without trailing slash.

**Comment by markseuffert** [comment]
You are right, location arguments work only for directory locations and not for page locations. Don't remember what the exact reason is, I think it's a technical limitation, Datenstrom Yellow wants URLs to be dynamically and statically generated.

**Comment by schulle4u** [comment]
Yes, this explains why I don't have much luck. I'll close this for now and rethink my solution. Thanks anyway. 😀
